docs/developers.rst

```diff
@@ -0,0 +1,56 @@
+
+More information on checking the reference simulations can be found in the README in the ``reference_simulations`` directory in the repository.
+
+We will periodically issue a new set of reference simulations as ``pyuvsim`` gains more capabilities.
+
+For more details, see `reference_simulations/README.md <https://github.com/RadioAstronomySoftwareGroup/pyuvsim/tree/master/reference_simulations>`_.
+
+Benchmarking
+------------
+
+The ``benchmarking`` directory contains tools to test the runtime and memory usage of large simulations. There is no requirement to check benchmarks for pull requests, but it's a good idea to make sure changes don't drastically alter the runtime. The file BENCHMARKS.log keeps a record of performance over time.
+
+The README file in the ``benchmarking`` directory gives more details on how to do benchmarking.
+
+Note that the benchmarking scripts are designed only for SLURM systems.
+
+For more details, see `benchmarking/README.md <https://github.com/RadioAstronomySoftwareGroup/pyuvsim/tree/master/benchmarking>`_.
+
+Parallelization
+---------------
+
+Simulations are massively parallelized through a combination of MPI and the use of ``numpy``/``scipy`` internal threading. Given ``Npus`` MPI processes, and ``Nsrcs``, ``Nbls``, ``Ntimes``, ``Nfreqs`` sources, baselines, times, and frequencies (respectively), the choice of splitting the various axes goes like this:
+
+    1. If ``Npus`` < ``Nsrcs`` and ``Npus > Nbls * Ntimes * Nfreqs``:
+           a. The source axis will be split across MPI processes.
+           b. Each process will receive all baselines, times, and frequencies.
+    2. Otherwise:
+           a. Each process will receive all sources, and times/frequencies/baselines will be split among MPI processes.
+           b. If it is estimated that loading all sources on each process simultaneously will overrun the available memory, then the sources will be split into chunks for processing.
+
+In each case, the source axis is handled through ``numpy``'s threading. It is recommended that jobs with especially large source axes be given more cores per MPI process (in SLURM, for instance, this is set by the ``--cpus-per-task`` option in ``sbatch`` or ``srun``). Usually around 2 to four cpus per process is sufficient. For large numbers of times/baselines/frequencies, however, running with more MPI processes offers a better speedup.
+
+The source array is initially shared in immutable shared memory, and parts of it are copied for usage within each MPI process. Likewise, UVBeam-class beam objects are loaded on the root process only, and broadcast using shared memory. These measures prevent large data arrays from being copied over ``Npus`` times, which can cause an unacceptable memory bloat.
```

Comment:
This section is super useful, but it feels more like it is for users than for developers. Maybe there's a better place to put it?

Reply:
That's a fair point. I moved it over to `usage.rst`.